src/main/java/info/mywinecellar/config/WebSecurityConfig.java

```diff
@@ -25,6 +25,9 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
         // Public access to login, landing, and error pages
         http.authorizeRequests().antMatchers("/", "/login", "/errorpage").permitAll();
 
+        // Logged user edit access
+        http.authorizeRequests().antMatchers(HttpMethod.POST, "/country/*/edit", "/grape/*/edit", "/area/*/edit").hasRole("USER");
```

Comment:
Great thanks!

We should add /region/*/edit to this as well.  I'm assuming you've tested locally that this works?

Reply:
This is really helpful @matiaspompilio thanks for the work

Reply:
@pauldpearson Yes, I have tested it locally and is working as expected. Adding region as well

Reply:
Done @pauldpearson

Reply:
Awesome, nice work, and yes please!

Reply:
@pauldpearson Done! Please review squashed commit. Thank you!